name: "jackal"
N: 30
integrator_step: 0.1
n_discs: 1

solver_settings:
  # solver: "acados" # acados or forces
  solver: "forces" # acados or forces
  floating_license: false
  enable_timeout: true
  init: 2 # 0 = cold start, 1 = centerer start, 2 = warm start with the selected primal variables
  use_sqp: false

recording:
  enable: false
  # folder: "/home/r2c1/Documents/publications/multi-mpc-2023/results/data"
  folder: "/home/r2c1/tmpc_revision/experiments/data"
  file: "GMPCC"
  # file: "LMPCC"
  timestamp: true
  num_experiments: 105

debug_limits: false
debug_output: false

enable_output: true
control_frequency: 20
deceleration_at_infeasible: 3.0
max_obstacles: 5
robot_radius: 0.325
robot:
  length: 0.65
  width: 0.65
  com_to_back: 0.0
obstacle_radius: 0.35

linearized_constraints:
  add_halfspaces: 0 #2  # For road constraints

scenario_constraints:
  parallel_solvers: 1

road:
  two_way: false
  width: 7.0

shift_previous_solution_forward: false

contouring:
  num_segments: 3
  preview: 0.0
  add_road_constraints: true

t-mpc:
  use_t-mpc++: true
  enable_constraints: true

probabilistic:
  enable: true
  risk: 0.05
  propagate_uncertainty: true

weights:
  goal_weight: 10.
  velocity: 0.3 
  acceleration: 0.25 #0.15
  angular_velocity: 0.5 #0.25 
  reference_velocity: 1.5 #2.0 
  contour: 0.05 #0.01 #0.05
  preview: 0.0
  lag: 0.1
  slack: 10000.

visualization:
  draw_every: 5 # stages